### PR TITLE
Make Stackdriver main collector more library-friendly

### PR DESCRIPTION
**Pull request by thepalbi**
We would like to use stackdriver_exporter as a library. The project is almost usable as a library as it is, though inside the main collectors code (`MonitoringCollector`) some CLI flag are used from the global context. 

This PR does a couple of changes to make `MonitoringCollector` more suitable for library use:
- Moves all `kingpin` CLI flags in a same file `flags.go`. This cleans the package `collectors`, making it only content the actual collector's code.
- Creates a new type `MonitoringCollectorOptions`, which will be used to send the newly created collector all required settings.
- Moves the parsing of `monitoring.metrics-type-prefixes`and `monitoring.filters` to the `main` package. Also for cleaning up the `collectors` package.

### CODE REVIEW

**Review by SuperQ — COMMENTED** [review]
Seems reasonable to me. Minor comment about the gitignore.

**Comment by SuperQ** [comment]
We've talked about making more of our exporter collector packages interchangeable to allow more library use. We're trying to come up with a registration/config pattern for this.

If you're interested in contributing, I would love to hear your thoughts.

File: .gitignore [hunk 1]
```diff
@@ -6,3 +6,6 @@
 *.test
 *-stamp
 /vendor
+
+# IDE Specific ignores
+/.idea
```
[the remark this answers is not in the document]
Comment (thepalbi) [reply .gitignore]: Thanks @SuperQ for the review. Just pushed a commit reverting that change 😀

**Comment by thepalbi** [comment]
> We've talked about making more of our exporter collector packages interchangeable to allow more library use. We're trying to come up with a registration/config pattern for this.
> 
> If you're interested in contributing, I would love to hear your thoughts.

Definetely, feel free to tag me in any issue/thread were you are discussing. More than happy to contribute 😀

**Review by SuperQ — APPROVED** [review]
LGTM, Thanks!

**Comment by SuperQ** [comment]
I didn't find an issue for it, so I opened https://github.com/prometheus/exporter-toolkit/issues/92.

**Pull request merged**